examples/TransactionGenerator.hs

```diff
@@ -241,178 +263,196 @@ scriptConfigParser = id
+  (cid, transaction) <- generateTransaction
+  (timeTaken, requestKeys) <- measureDiffTime (sendTransaction cid transaction)
+  lift . logg Info $ toLogMessage ("Sent transaction with request keys: " <> sshow requestKeys :: Text)
   when mtime $
-    lift $ logg Info (toLogMessage (("Sending a transaction (with request keys: " <> sshow requestKeys <> ") took: " <> sshow timeTaken) :: Text))
+    lift $ logg Info $ toLogMessage ("Sending a transaction (with request keys: " <> sshow requestKeys <> ") took: " <> sshow timeTaken :: Text)
   count <- use gsCounter
   gsCounter += 1
-  lift $ logg Info (toLogMessage (("Transaction count: " <> sshow count) :: Text))
-  forkedListens requestKeys
-  loop measure
+  lift . logg Info $ toLogMessage ("Transaction count: " <> sshow count :: Text)
+  case requestKeys of
+    Left servantError -> lift . logg Error $ toLogMessage (sshow servantError :: Text)
+    Right rks -> forkedListens tq cid rks
+  logs <- liftIO . atomically $ flushTQueue tq
+  lift $ traverse_ (logg Info . toLogMessage) logs
+  loop tq measure
 
 forkedListens
   :: forall m. (MonadIO m, MonadLog SomeLogMessage m, MonadBaseControl IO m)
-  => Either ClientError RequestKeys
+  => TQueue Text
+  -> ChainId
+  -> RequestKeys
   -> TXG m ()
-forkedListens requestKeys = do
-  err <- traverse (traverse forkedListen) (traverse _rkRequestKeys requestKeys)
-  case sequence err of
-    Left servantError -> lift $ logg Error (toLogMessage (sshow servantError :: Text))
-    Right _ -> return ()
+forkedListens tq cid (RequestKeys rks) = do
+  TXGConfig _ _ ce v <- ask
+  liftIO $ traverse_ (forkedListen ce v) rks
   where
-    forkedListen :: RequestKey -> TXG m ()
-    forkedListen requestKey = do
-        liftIO $ print requestKey
-        clientEnv <- view genClientEnv
-        chain <- view genChainId
-        version <- view genVersion
-        let listenerRequest = ListenerRequest requestKey
-        -- LoggerT has an instance of MonadBaseControl. Also, there is a
-        -- function from `monad-control` which enables you to lift forkIO. The
-        -- extra lift at the end is to get the entire computation back into the
-        -- TXG transformer.
-        void $ lift $ liftBaseDiscard forkIO $ do
-          (time,response) <- liftIO $ measureDiffTime (runClientM (listen version chain listenerRequest) clientEnv)
-          liftIO $ print response
-          -- withLabel ("component", "transaction-generator") $ -- add this back in later
-          logg Info $ toLogMessage (("It took " <> sshow time <> " seconds to get back the result.") :: Text)
-          -- withLabel ("component", "transaction-generator") $ -- add this back in later
-          logg Info $ toLogMessage (("The associated request is " <> sshow requestKey <> "\n" <> sshow response) :: Text)
+    forkedListen :: ClientEnv -> ChainwebVersion -> RequestKey -> IO ()
+    forkedListen ce v rk = do
+      print rk  -- TODO Should this be printed or logged?
```

Comment:
I think I had left that there by accident. I think it should be logged, so along with the metrics collected below.